knn/knn.py

```diff
@@ -644,7 +644,14 @@ def extract_from_zarr(zarr_path, aoi_path, target_path, open_chunks=-1):
             & (dataset.lon <= maxx + width)
             & (dataset.lat >= miny - height)
             & (dataset.lat <= maxy + height), drop=True)
-        dataset.to_netcdf(target_path)
+
+        try:
+            dataset.to_netcdf(target_path)
+        except ValueError:
```

Comment:
I could imagine a time when there might be a `ValueError` here that's unrelated to the datetime conversion. So instead of the `try/except` pattern, it might be good to check the type of `dataset.time`, and if it's not already `datetime64`, then do the conversion. That way we also guarantee that all netcdfs that we write use the same time time datatype.